src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -0,0 +1,454 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
```

Comment:
Thanks for writing a bunch of tests!

I'm curious; did you look at Mono to see what tests it has?  If it has good ones, would you want to look at porting them in addition to these?